File: src/slang/parse/SkipParselet.java
package slang.parse;

import slang.lex.Token;

/**
 * Just do nothing and return null. Lowest precedence.
 */
public class SkipParselet implements PrefixParselet, InfixParselet {
    @Override
    public Expr parse(Parser parser, Token token) {
        return null;
    }

    @Override
    public Expr parse(Parser parser, Expr left, Token token) {
        return null;
    }

    @Override
    public int getPrecedence() {
        return 0;
    }
}

File: src/slang/parse/SkipParselet.scala
package slang.parse

import slang.lex.Token

/**
 * Just do nothing and return null. Lowest precedence.
 */
class SkipParselet extends PrefixParselet with InfixParselet {
    override def parse(parser: Parser, token: Token): Expr = null
    override def parse(parser: Parser, left: Expr, token: Token): Expr = null
    override def getPrecedence: Int = -1
}
